app/controllers/api/v1/cards_controller.rb

```diff
@@ -15,13 +15,40 @@ def index
   end
 
   def search
+    hash = nil
+
     if valid_search_params?
       cards = Card.search(format_search_params)
-      render json: CardSerializer.new(cards)
+      hash = CardSerializer.new(cards)
+      hash = JSON.parse(hash.to_json, symbolize_names: true)
+      hash[:error] = []
     else
-      render json: { error: find_invalid_search_keys },
-             status: :bad_request
+      invalid = find_invalid_search_keys
+      valid_params = format_search_params
+
+      invalid.each do |key|
+        valid_params.delete(key)
+      end
+
+      if valid_params != {}
+        cards = Card.search(valid_params)
+
+        hash = JSON.parse(
+          CardSerializer.new(cards).to_json,
+          symbolize_names: true
+        )
+      else
+        hash = { data: [] }
+      end
+
+      hash[:error] = []
+
+      invalid.each do |key|
+        hash[:error] << "#{key} is an invalid search parameter"
```

Comment:
I think a custom serializer may have been an easier way to handle returning the errors but very nice job figuring this out!